src/blob_seq.rs

```diff
@@ -261,6 +261,49 @@ impl BlobSeq {
         })
     }
 
+    /// For a value `x`, returns the immedate next [`BlobSeq`] `y > x`, such that
+    /// there is **no** `intermediate` for which `x < intermediate < y` holds.
+    pub fn immediate_successor(&self) -> BlobSeq {
+        Self::from(
+            self.components()
+                .map(|slice| slice.to_vec())
```

Comment:
Pretty sure this can be done without all these allocations. But this is not the time. Maybe worth a todo!?

Reply:
Briefly took a look at this, and yeah. You're right.

I skipped actually doing this once I realized I'd need to dive into the data layout for `BlobSeq` first. So I hope it's okay if I defer this to you.

Will leave some TODOs though.